padguidedb/padguidedb.py

```diff
@@ -111,40 +118,74 @@ async def searchdungeon(self, ctx, *, search_text):
+            msg = 'Results\n' + json.dumps(results, indent=2, ensure_ascii=False)
+            await ctx.send(inline(sql))
+            for page in pagify(msg):
+                await ctx.send(box(page))
 
-    @padguidedb.command()
-    @checks.is_owner()
-    async def setimageupdatefile(self, ctx, *, image_update_file):
-        """Set the image update file."""
-        self.settings.setImageUpdateFile(image_update_file)
-        await ctx.tick()
+    @series.command(name="add")
+    async def series_add(self, ctx, *, elements: TokenConverter):
+        """Add"""
+        if not all(x in SERIES_KEYS for x in elements):
+            await ctx.send("Invalid key.  Valid keys are: `{}`".format("` `".join(SERIES_KEYS)))
+            return
 
-    @padguidedb.command()
-    @checks.is_owner()
-    async def setpythonexecutable(self, ctx, *, python_executable):
-        """Set the python executable file."""
-        self.settings.setPythonExecutable(python_executable)
-        await ctx.tick()
+        EXTRAS = {}
+        with self.get_cursor() as cursor:
+            cursor.execute("SELECT MAX(series_id) FROM series")
+            max_val = cursor.fetchall()[0]['MAX(series_id)']
+            EXTRAS['series_id'] = max_val + 1
+        EXTRAS['tstamp'] = int(datetime.now().timestamp())
+        elements = {**SERIES_KEYS, **EXTRAS, **elements}
+        
+        key_infix = ", ".join(elements.keys())
+        value_infix = ", ".join("%s" for v in elements.values())
+        with self.get_cursor() as cursor:
+            sql = ('INSERT INTO series ({})'
+                   ' VALUES ({})').format(key_infix, value_infix)
+            affected = cursor.execute(sql, (*elements.values(),))
+            await ctx.send(inline(cursor._executed))
+            await ctx.send("{} row(s) added.".format(affected))
+
+    @series.command(name="edit")
+    async def series_edit(self, ctx, series_id: int, *, elements: TokenConverter):
+        """Edit"""
+        if not all(x in SERIES_KEYS for x in elements):
+            await ctx.send("Invalid key.  Valid keys are: `{}`".format("` `".join(SERIES_KEYS)))
+            return
 
-    @padguidedb.command()
-    @checks.is_owner()
-    async def setuserinfo(self, ctx, server: str, user_uuid: str, user_intid: str):
-        """Set the dungeon script."""
-        self.settings.setUserInfo(server, user_uuid, user_intid)
-        await ctx.tick()
+        replacement_infix = ", ".join(["{} = %s".format(k) for k in elements.keys()])
+        with self.get_cursor() as cursor:
+            sql = ('UPDATE series'
+                   ' SET {}'
+                   ' WHERE series_id = %s').format(replacement_infix)
+            affected = cursor.execute(sql, (*elements.values(),) + (series_id,))
+            await ctx.send(inline(cursor._executed))
+            await ctx.send("{} row(s) changed.".format(affected))
+
+    @series.command(name="delete")
+    async def series_delete(self, ctx, series_id: int):
```

Comment:
thing to keep in mind in the future: 

what happens if you try to delete a series where monsters are still assigned to the id? is that a okay? should we search all the monsters and remove those links as well? (i'm more in the camp of having weak references and leaving it up to manual cleanup, rather than strong foreign keys)

Reply:
I thought they were linked via foreign keys, but they are not.  This would be nice to add tbh, but right now it will break things if it's deleted.
lazy solution: make delete owner only (?)